src/App.tsx

```diff
@@ -49,34 +59,30 @@ export const App: React.FC = () => {
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector value={author} onChange={setAuthor} />
+                <UserSelector value={author} onChange={handleSetAuthor} />
               </div>
 
               <div className="block" data-cy="MainContent">
                 {!author && <p data-cy="NoSelectedUser">No user selected</p>}
-
-                {author && !loaded && <Loader />}
-
-                {author && loaded && hasError && (
+                {author && !isLoading && <Loader />}
+                {author && isLoading && error && (
```

Comment:
Would be great to create variables for such complicated condition to improve readability